src/client/filter.rs

```diff
@@ -0,0 +1,264 @@
+pub enum ObCompareOperator {
+    LessThan = 0,
+    GreaterThan = 1,
+    LessOrEqualThan = 2,
+    GreaterOrEqualThan = 3,
+    NotEqual = 4,
+    Equal = 5,
+    IsNull = 6,
+    IsNotNull = 7,
+}
+
+impl ObCompareOperator {
+    pub fn string(&self) -> &'static str {
+        match self {
+            ObCompareOperator::LessThan => "<",
+            ObCompareOperator::GreaterThan => ">",
+            ObCompareOperator::LessOrEqualThan => "<=",
+            ObCompareOperator::GreaterOrEqualThan => ">=",
+            ObCompareOperator::NotEqual => "!=",
+            ObCompareOperator::Equal => "=",
+            ObCompareOperator::IsNull => "IS",
+            ObCompareOperator::IsNotNull => "IS_NOT",
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum FilterOp {
+    And = 0,
+    Or = 1,
+}
+
+pub struct ObTableFilterList {
+    pub op: FilterOp,
+    pub filters: Vec<Box<dyn Filter>>,
+}
+
+#[macro_export]
+macro_rules! filter_list {
+    ($op:expr, $($filter:expr),+ $(,)?) => {
+        ObTableFilterList {
+            op: $op,
+            filters: vec![$(Box::new($filter) as Box<dyn Filter>),+],
+        }
+    };
+}
+
+impl Filter for ObTableFilterList {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn string(&self) -> String {
+        let string_op = match self.op {
+            FilterOp::And => " && ",
+            FilterOp::Or => " || ",
+        };
+
+        // Use an iterator with map and collect to efficiently concatenate strings
+        self.filters
+            .iter()
+            .map(|filter| {
+                let filter_string = filter.string();
+                if filter.as_any().is::<ObTableValueFilter>() {
+                    filter_string
+                } else {
+                    format!("({})", filter_string)
+                }
+            })
+            .collect::<Vec<_>>()
+            .join(string_op)
+    }
+
+    fn concat_string(&self, filter_string: &mut String) {
+        let string_op = match self.op {
+            FilterOp::And => " && ",
+            FilterOp::Or => " || ",
+        };
+
+        for (i, filter) in self.filters.iter().enumerate() {
+            if i != 0 {
+                filter_string.push_str(string_op);
+            }
+            if filter.as_any().is::<ObTableValueFilter>() {
+                filter.concat_string(filter_string);
+            } else {
+                filter_string.push('(');
+                filter.concat_string(filter_string);
+                filter_string.push(')');
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ObTableValueFilter {
+    pub op: ObCompareOperator,
+    pub column_name: String,
+    pub value: String,
+}
+
+#[macro_export]
+macro_rules! value_filter {
+    ($op:expr, $column_name:expr, $value:expr) => {
+        ObTableValueFilter {
+            op: $op,
+            column_name: $column_name.to_string(),
+            value: $value.to_string(),
+        }
+    };
+}
```

Comment:
I guess no need to define such a macro, and a method named after `ObTableValueFilter::new()` is enough.

Reply:
Fixed, add `new()` for ObTableValueFilter.
Also, add `new()` for ObTableFilterList. For convenience, we add `add_filter()`/`add_filters()` for users to push other filters into the filter list.